src/redux/pattern/pattern.sagas.js

```diff
@@ -0,0 +1,122 @@
+import { takeEvery, call, put } from 'redux-saga/effects';
+import { push } from 'connected-react-router';
+import {
+  setPatterns,
+  setPatternLoading,
+  setPattern,
+  setPatternError
+} from './pattern.actions';
+
+import {
+  getAllPatterns,
+  deletePattern,
+  createPattern,
+  updatePattern,
+  getPatternById
+} from './pattern.operations';
+
+import { config, routes } from '../../configs';
+
+import {
+  setSnackBarSeverity,
+  setSnackBarStatus,
+  setSnackBarMessage
+} from '../snackbar/snackbar.actions';
+import {
+  GET_PATTERNS,
+  DELETE_PATTERN,
+  ADD_PATTERN,
+  UPDATE_PATTERN,
+  GET_PATTERN
+} from './pattern.types';
+
+const {
+  SUCCESS_ADD_STATUS,
+  SUCCESS_DELETE_STATUS,
+  SUCCESS_UPDATE_STATUS
+} = config.statuses;
+
+function* handlePatternsLoad() {
+  try {
+    yield put(setPatternLoading(true));
+    const patterns = yield call(getAllPatterns, null);
+    yield put(setPatterns(patterns));
+    yield put(setPatternLoading(false));
+  } catch (error) {
+    yield call(handlePatternError, error);
+  }
+}
+
+function* handlePatternLoad({ payload }) {
+  try {
+    yield put(setPatternLoading(true));
+    const pattern = yield call(getPatternById, payload);
+    console.log(pattern);
+    yield put(setPattern(pattern));
+    yield put(setPatternLoading(false));
+  } catch (error) {
+    yield call(handlePatternError, error);
+  }
+}
+
+function* handleAddPattern({ payload }) {
+  try {
+    yield put(setPatternLoading(true));
+    yield call(createPattern, payload);
+    const patterns = yield call(getAllPatterns, null);
+    yield put(setPatterns(patterns));
+    yield put(setSnackBarSeverity('success'));
+    yield put(setSnackBarMessage(SUCCESS_ADD_STATUS));
+    yield put(setSnackBarStatus(true));
+    yield put(push(routes.pathToPatterns));
+  } catch (error) {
+    yield call(handlePatternError, error);
+  }
+}
+
+function* handlePatternDelete({ payload }) {
+  try {
+    yield put(setPatternLoading(true));
+    yield call(deletePattern, payload);
+    const patterns = yield call(getAllPatterns, null);
+    yield put(setPatterns(patterns));
+    yield put(setPatternLoading(false));
+    yield put(setSnackBarSeverity('success'));
+    yield put(setSnackBarMessage(SUCCESS_DELETE_STATUS));
+    yield put(setSnackBarStatus(true));
+  } catch (error) {
+    yield call(handlePatternError, error);
+  }
+}
+
+function* handlePatternUpdate({ payload }) {
+  const { id, pattern } = payload;
+  try {
+    yield put(setPatternLoading(true));
+    yield call(updatePattern, id, pattern);
+    const patterns = yield call(getAllPatterns, null);
```

Comment:
There is no need to fetch all patterns after updating, you need only to update the object of the updated pattern in the redux and that's all, you don't need any unnecessary HTTP request to the backend

Reply:
fixed